docs/using.md

```diff
@@ -54,23 +54,47 @@ To create a Task, you must:
 Each container image used as a step in a [`Task`](#task) must comply with a specific
 contract.
 
-* [The `entrypoint` of the image will be ignored](#step-entrypoint)
+When containers are run in a `Task`, the `entrypoint` of the container will be
+overwritten with a custom binary that redirects the logs to a separate location
+for aggregating the log output. Due to this, when the `entrypoint` is overridden
+the controller will make a request to the image registry to try and get the
+metadata for that image, which will contain the built in `entrypoint`. As long
+as the entrypoint isn't explicitly overridden, this lookup will happen.
 
-For example, in the following Task the images, `gcr.io/cloud-builders/gcloud`
-and `gcr.io/cloud-builders/docker` run as steps:
+Due to this metadata lookup, this means that if you use a private image as a
+step inside a `Task`, the build-pipeline controller needs to be able to access
+that registry. The simplest way to accomplish this is to add a `.docker/config.json`
+at `$HOME/.docker/config.json`, which will then be used by the controller when
+performing the lookup
+
+
+For example, in the following Task with the images, `gcr.io/cloud-builders/gcloud`
+and `gcr.io/cloud-builders/docker`, the entrypoint would be resolved from the
+registry, resulting in the tasks running `gcloud` and `docker` respectively.
 
 ```yaml
 spec:
   buildSpec:
     steps:
     - image: gcr.io/cloud-builders/gcloud
-      command: ['gcloud']
       ...
     - image: gcr.io/cloud-builders/docker
-      command: ['docker']
```

Comment:
i still think that the first example ppl encounter here should include `command`, at the moment it seems like specifying `command` is the exception.

i think the docs should be like:

1. here is how to create a Task with command
2. oh but if you really want to use entrypoint, here is how that works, will xyz caveats